lib/fluent/plugin/out_split.rb

```diff
@@ -44,9 +50,18 @@ def split_message(message)
       data = {}
       message.split(@sep_regex).each do |e|
         matched = @format_regex.match(e) or next
-        data.store(matched['key'], matched['value'])
+        @store_fun.call(data, matched['key'], matched['value'])
```

Comment:
Thanks for clarification. I haven't care the performance here for now, but that's right what you think. Please keep it.